### DISCUSSION TITLE
Rendering to multiple outputs

### DISCUSSION THREAD

**Discussion by andrewpbray** [quarto-dev/quarto-cli#2021] [Q&A] [answered]
Is it possible to render a single qmd file into multiple simultaneous formats each render?

I have a website with certain pages that would be useful to automatically render to html (for the website) and to pdf for the option to download a pdf version. I thought about setting it up as a pre- or post-render script but I haven't been able to get it working yet (it also had me wondering about getting trapped in infinite render loops).

In an ideal world this would be a qmd-specific option through the yaml.

**Comment by jjallaire** [comment, marked as answer]
This should work if you list _both_ `html` and `pdf` as target formats for those documents. The current heuristic is that any document without a `format` gets the website default and any document with a `format` will render exactly the formats listed (which allows you to render to _just_ PDF, Revealjs, Markdown, etc. -- which is probably the more common case). So TLDR this should do it:

```yaml
---
title: "Document"
format:
    html: default 
    pdf: default
---
```

**Reply by andrewpbray** [reply]
Ah, wonderful! Ok, that works as perfectly in a single qmd setting, but for the website it has some unexpected behavior. If I toggle a single blog post to render as both HTML and pdf (with shared metadata in `_quarto.yml`), when I render the site and look into the `_site` dir, it looks like I have HTML and pdf copies of every qmd file in the website. All of the other qmds in the website have no specified format.

I'll keep tinkering around with this and pull together a minimal example repo to link here I can't figure it out.

**Comment by jjallaire** [comment]
You want it to be more scoped that that -- i.e. just put it in the metadata of the individual file(s) or use a directory level `_metadata.yml`

**Reply by andrewpbray** [reply]
Got it, that makes sense - thanks!
